protos/sigstore_rekor.proto

```diff
@@ -0,0 +1,109 @@
+// Copyright 2022 The Sigstore Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+syntax = "proto3";
+package dev.sigstore.rekor.v1;
+
+option go_package = "github.com/sigstore/proto/rekor/v1";
+option java_package = "dev.sigstore.proto.rekor.v1";
+option java_multiple_files = true;
+option java_outer_classname = "RekorProto";
+
+// KindVersion contains the entry's kind and api version.
+message KindVersion {
+        // Kind is the type of entry being stored in the log.
+        // See here for a list: https://github.com/sigstore/rekor/tree/main/pkg/types
+        string kind = 1;
+        // The specific api version of the type.
+        string version = 2;
+}
+
+// The checkpoint contains a signature of the tree head (root hash),
+// size of the tree, the transparency log's unique identifier (log ID),
+// hostname and the current time.
+// The result is a string, the format is described here
+// https://github.com/transparency-dev/formats/blob/main/log/README.md
+// The details are here https://github.com/sigstore/rekor/blob/main/pkg/util/signed_note.go#L114.
+// The signature has the same format as
+// TransparencyLogEntry.inclusion_proof. See below for more details.
+message Checkpoint {
+        string envelope = 1;
+}
+
+// InclusionProof is the proof returned from the transparency log. Can
+// be used for on line verification against the log.
+message InclusionProof {
+        // The index of the entry in the log.
+        int64 log_index = 1;
+        // The hash digest stored at the root of the merkle tree at the time
+        // the proof was generated.
+        bytes root_hash = 2;
+        // The size of the merkle tree at the time the proof was generated.
+        int64 tree_size = 3;
+        // A list of hashes required to compute the inclusion proof, sorted
+        // in order from leaf to root.
```

Comment:
I think leaf and root aren't included in the proof - @asraa right?
I would maybe omit anything about sorting, it must be correctly sorted by definition

Reply:
This text is a raw copy from the swagger API document. But I agree, we should get the wording correct.

Reply:
It doesn't include the root hash, I think it may include the leaf - `curl "https://rekor.sigstore.dev/api/v1/log/proof?firstSize=20000&lastSize=200001&treeID=3904496407287907110" | jq`

Reply:
It does not include the LEAF hash, that should be calculated by the entry body.

It DOES contain the root hash.

Reply:
Hayden: `/log/proof` serves a consistency proof, so what you're seeing is the proof list for consistency between the specified first and last size,

Reply:
IMO the wording is "correct", the ordering is from leaf to root, and ordering doesn't mean bounds. But I agree on clarity including that root hash is included and leaf is not! `(leaf, root]`, and that `checkpoint` is a signature on the root hash.

Reply:
EDIT (sorry): It doesn't include the root hash, the root hash is taken from the `root_hash` field above, it's separate.

Reply:
EDIT (sorry!): It doesn't include the root hash, the root hash is taken from the `root_hash` field above, it's separate.

Reply:
EDIT (sorry!): It doesn't include the root hash, the root hash is taken from the `root_hash` field above, it's separate.

Reply:
EDIT (sorry!): It doesn't include the root hash, the root hash is taken from the `root_hash` field above, it's separate.